[Localization] Modification some code about battle stat for localization

Ohmry opened the pull request:
<!-- Make sure the title includes categorization (i.e. [Bug], [QoL], [Localization]) -->
<!-- Make sure that this PR is not overlapping with someone else's work -->
<!-- Please try to keep the PR self-contained (and small) -->

## What are the changes?
<!-- Summarize what are the changes from a user perspective on the application -->
Modification some code about battle stat for localization.

## Why am I doing these changes?
<!-- Explain why you decided to introduce these changes -->
<!-- Does it come from an issue or another PR? Please link it -->
<!-- Explain why you believe this can enhance user experience -->
The message has been shown only english about battle stat such as ATK, DEF. so that it needs to modify that can be localized by translators I think.

## What did change?
<!-- Explicitly state what are the changes introduced by the PR -->
<!-- You can make use of a comparison between what was the state before and after your PR changes -->
- Modification return value of `getBattleStatName`, `getBattleStatLevelChangeDescription` method. it returns localized value used to i18next not English.
- Created `battle-stat.ts` in each locale folders for localizing by translators, they can translate messages about battle stat from this file.
- Created `battle-stat.test.ts` for test localization of battle stat.

### Screenshots/Videos
<!-- If your change is changing anything on the user experience, please provide visual proofs of it -->
<!-- Please take screenshots/videos before and after your changes, to show what is brought by this PR -->

#### Korean
<img width="828" alt="스크린샷 2024-06-09 오후 1 46 59" src="https://github.com/pagefaultgames/pokerogue/assets/13056250/9989e5c7-ae19-46a1-89db-098232450cc1">

#### Other languages
<img width="908" alt="스크린샷 2024-06-09 오후 1 45 41" src="https://github.com/pagefaultgames/pokerogue/assets/13056250/1f710daa-3b8b-4c80-8606-5df14f530429">

#### Result of test
<img width="665" alt="image" src="https://github.com/pagefaultgames/pokerogue/assets/13056250/9e8ecc31-7268-479d-8517-2dc192659aaf">

## How to test the changes?
<!-- How can a reviewer test your changes once they check out on your branch? -->
<!-- Did you just make use of the `src/overrides.ts` file? -->
<!-- Did you introduce any automated tests? -->
<!-- Do the reviewer need to do something special in order to test your change? -->

- Check out this PR and you can check when you use skill to change battle stat such as growl.
- In NPM, you can test using vitest, `npm run test battle-stat.test.ts`

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

CodeTappert commented:
The stat ones already exist in pokemon-info.ts 

Maybe use those instead? We dont need to have them localized twice

Ohmry commented:
> The stat ones already exist in pokemon-info.ts
> 
> Maybe use those instead? We dont need to have them localized twice

You're right, I missed that. I modified localized key about battle stat that getting from pokemon-info not battle-stat. and I think battle-stat.ts is not necessary for storing one section. so that I moved level change description to battle.ts from battle-stat.ts. and I deleted all battle-stat.ts file too.

done test too.
![image](https://github.com/pagefaultgames/pokerogue/assets/13056250/e4d1afcf-2f95-4038-b68c-6f79c0042056)

CodeTappert commented:
> > The stat ones already exist in pokemon-info.ts
> > Maybe use those instead? We dont need to have them localized twice
> 
> You're right, I missed that. I modified localized key about battle stat that getting from pokemon-info not battle-stat. and I think battle-stat.ts is not necessary for storing one section. so that I moved level change description to battle.ts from battle-stat.ts. and I deleted all battle-stat.ts file too.
> 
> done test too. ![image](https://private-user-images.githubusercontent.com/13056250/337961128-e4d1afcf-2f95-4038-b68c-6f79c0042056.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc5MjUzNDMsIm5iZiI6MTcxNzkyNTA0MywicGF0aCI6Ii8xMzA1NjI1MC8zMzc5NjExMjgtZTRkMWFmY2YtMmY5NS00MDM4LWI2OGMtNmY3OWMwMDQyMDU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA5VDA5MjQwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjODFmNmQ5NzA4MTNiYTJiMDNjNzFhZTE2MGY3ZTA5MTczOGUwZGYzZGZjNjAyYjAzMDhlOGJjZmVhMWE1MmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.slipPagaWs2ygQF2A82dUPAuDiP2gzbyFZHRegp1dtY)

You did not delete the german battle-stat.ts

Ohmry commented:
> > > The stat ones already exist in pokemon-info.ts
> > > Maybe use those instead? We dont need to have them localized twice
> > 
> > 
> > You're right, I missed that. I modified localized key about battle stat that getting from pokemon-info not battle-stat. and I think battle-stat.ts is not necessary for storing one section. so that I moved level change description to battle.ts from battle-stat.ts. and I deleted all battle-stat.ts file too.
> > done test too. ![image](https://private-user-images.githubusercontent.com/13056250/337961128-e4d1afcf-2f95-4038-b68c-6f79c0042056.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc5MjUzNDMsIm5iZiI6MTcxNzkyNTA0MywicGF0aCI6Ii8xMzA1NjI1MC8zMzc5NjExMjgtZTRkMWFmY2YtMmY5NS00MDM4LWI2OGMtNmY3OWMwMDQyMDU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA5VDA5MjQwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjODFmNmQ5NzA4MTNiYTJiMDNjNzFhZTE2MGY3ZTA5MTczOGUwZGYzZGZjNjAyYjAzMDhlOGJjZmVhMWE1MmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.slipPagaWs2ygQF2A82dUPAuDiP2gzbyFZHRegp1dtY)
> 
> You did not delete the german battle-stat.ts

I deleted and push again, thanks.

Greenlamp2 commented:
Could you move your unit test into test/lokalisation please? thanks !

Ohmry commented:
> Could you move your unit test into test/lokalisation please? thanks !

Sure, I did c8f8114

CodeTappert commented:
> Could you move your unit test into test/lokalisation please? thanks !

Isnt the folder named wrong? Or is my english incorrect?

Greenlamp2 commented:
> > Could you move your unit test into test/lokalisation please? thanks !
> 
> Isnt the folder named wrong? Or is my english incorrect?

no, you're right, I'm used to using a service called "Lokalise" to manage translations, and the more I use it, the more it's changed in my mind how it's written ahah

returntoice reviewed (approved):
Okay enough for Korean translation.

Tempo-anon commented on src/data/battle-stat.ts:
```diff
@@ -34,30 +36,30 @@ export function getBattleStatLevelChangeDescription(levels: integer, up: boolean
     case 1:
-      return "rose";
+      return i18next.t("battle:statIsRose");
     case 2:
-      return "sharply rose";
+      return i18next.t("battle:statIsSharplyRose");
```
nit: I'm not a fan of having `Is` in all of these. They can simply be `statRose` `statSharplyRose` `statRoseDrastically` `statWontGoAnyHigher` `statFell` `statHarshlyFell` `statSeverelyFell` and `statWontGoAnyLower`

Ohmry replied:
I will fix it and push again. and may I ask you check #1934? I merged commits that conflicted.

Ohmry replied:
I fixed. please check it.

Ohmry commented:
@Tempo-anon Hi, I resolved the conflicted. please confirm again.
